
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>cardea.fhir.Sequence module &#8212; Cardea 0.1.2.dev0 documentation</title>
    
  <link rel="stylesheet" href="../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../_static/basic.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
  <link rel="preload" as="script" href="../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/language_data.js"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.ico"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <nav class="navbar navbar-light navbar-expand-lg bg-light fixed-top bd-navbar" id="navbar-main">
<div class="container-xl">

    <a class="navbar-brand" href="../index.html">
    
      <p class="title">Cardea</p>
    
    </a>
    <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbar-menu" aria-controls="navbar-menu" aria-expanded="false" aria-label="Toggle navigation">
        <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar-menu" class="col-lg-9 collapse navbar-collapse">
      <ul id="navbar-main-elements" class="navbar-nav mr-auto">
        
        
        <li class="nav-item ">
            <a class="nav-link" href="../getting_started/index.html">Getting Started</a>
        </li>
        
        <li class="nav-item ">
            <a class="nav-link" href="../basic_concepts/index.html">Basic Concepts</a>
        </li>
        
        <li class="nav-item ">
            <a class="nav-link" href="../api_reference/index.html">API Reference</a>
        </li>
        
        <li class="nav-item ">
            <a class="nav-link" href="../community/index.html">Community</a>
        </li>
        
        <li class="nav-item ">
            <a class="nav-link" href="../history.html">Release Notes</a>
        </li>
        
        
      </ul>


      

      <ul class="navbar-nav">
        
          <li class="nav-item">
            <a class="nav-link" href="https://github.com/MLBazaar/Cardea" target="_blank" rel="noopener">
              <span><i class="fab fa-github-square"></i></span>
            </a>
          </li>
        
        
      </ul>
    </div>
</div>
    </nav>
    

    <div class="container-xl">
      <div class="row">
          
          <div class="col-12 col-md-3 bd-sidebar"><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">

    <div class="bd-toc-item active">
    
  
    <ul class="nav bd-sidenav">
        
        
        
        
        
        
        
        
        
        
        
      </ul>
  
  </nav>
          </div>
          

          
          <div class="d-none d-xl-block col-xl-2 bd-toc">
              

<nav id="bd-toc-nav">
    <ul class="nav section-nav flex-column">
    
    </ul>
</nav>


              
          </div>
          

          
          <main class="col-12 col-md-9 col-xl-7 py-md-5 pl-md-5 pr-md-4 bd-content" role="main">
              
              <div>
                
  <div class="section" id="module-cardea.fhir.Sequence">
<span id="cardea-fhir-sequence-module"></span><h1>cardea.fhir.Sequence module<a class="headerlink" href="#module-cardea.fhir.Sequence" title="Permalink to this headline">¶</a></h1>
<dl class="class">
<dt id="cardea.fhir.Sequence.Sequence">
<em class="property">class </em><code class="sig-prename descclassname">cardea.fhir.Sequence.</code><code class="sig-name descname">Sequence</code><span class="sig-paren">(</span><em class="sig-param">dict_values=None</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/cardea/fhir/Sequence.html#Sequence"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#cardea.fhir.Sequence.Sequence" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="cardea.fhir.fhirbase.html#cardea.fhir.fhirbase.fhirbase" title="cardea.fhir.fhirbase.fhirbase"><code class="xref py py-class docutils literal notranslate"><span class="pre">cardea.fhir.fhirbase.fhirbase</span></code></a></p>
<p>Raw data describing a biological sequence.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>resourceType</strong> – This is a Sequence resource</p></li>
<li><p><strong>identifier</strong> – A unique identifier for this particular sequence instance.
This is a FHIR-defined id.</p></li>
<li><p><strong>type</strong> – Amino Acid Sequence/ DNA Sequence / RNA Sequence.</p></li>
<li><p><strong>coordinateSystem</strong> – Whether the sequence is numbered starting at 0
(0-based numbering or coordinates, inclusive start, exclusive end) or
starting at 1 (1-based numbering, inclusive start and inclusive end).</p></li>
<li><p><strong>patient</strong> – The patient whose sequencing results are described by this
resource.</p></li>
<li><p><strong>specimen</strong> – Specimen used for sequencing.</p></li>
<li><p><strong>device</strong> – The method for sequencing, for example, chip information.</p></li>
<li><p><strong>performer</strong> – The organization or lab that should be responsible for this
result.</p></li>
<li><p><strong>quantity</strong> – The number of copies of the seqeunce of interest. (RNASeq).</p></li>
<li><p><strong>referenceSeq</strong> – A sequence that is used as a reference to describe
variants that are present in a sequence analyzed.</p></li>
<li><p><strong>variant</strong> – The definition of variant here originates from Sequence
ontology
([variant_of](<a class="reference external" href="http://www.sequenceontology.org/browser/current_svn/term/variant_of">http://www.sequenceontology.org/browser/current_svn/term/variant_of</a>)).
This element can represent amino acid or nucleic sequence
change(including insertion,deletion,SNP,etc.)  It can represent some
complex mutation or segment variation with the assist of CIGAR string.</p></li>
<li><p><strong>observedSeq</strong> – Sequence that was observed. It is the result marked by
referenceSeq along with variant records on referenceSeq. This shall
starts from referenceSeq.windowStart and end by
referenceSeq.windowEnd.</p></li>
<li><p><strong>quality</strong> – An experimental feature attribute that defines the quality of
the feature in a quantitative way, such as a phred quality score
([SO:0001686](http://www.sequenceontology.org/browser/current_svn/term/SO:0001686)).</p></li>
<li><p><strong>readCoverage</strong> – Coverage (read depth or depth) is the average number of
reads representing a given nucleotide in the reconstructed sequence.</p></li>
<li><p><strong>repository</strong> – Configurations of the external repository. The repository
shall store target’s observedSeq or records related with target’s
observedSeq.</p></li>
<li><p><strong>pointer</strong> – Pointer to next atomic sequence which at most contains one
variant.</p></li>
</ul>
</dd>
</dl>
<dl class="method">
<dt id="cardea.fhir.Sequence.Sequence.assert_type">
<code class="sig-name descname">assert_type</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/cardea/fhir/Sequence.html#Sequence.assert_type"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#cardea.fhir.Sequence.Sequence.assert_type" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks class values follow set possible enumerations.</p>
<dl class="field-list simple">
<dt class="field-odd">Raises</dt>
<dd class="field-odd"><p><strong>ValueError</strong> – An error occurs if an attribute of the fhir class
    does not match its possible enumerations.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="cardea.fhir.Sequence.Sequence.get_relationships">
<code class="sig-name descname">get_relationships</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/cardea/fhir/Sequence.html#Sequence.get_relationships"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#cardea.fhir.Sequence.Sequence.get_relationships" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns class relationships.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>A list of the class’s associated relationships.</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="cardea.fhir.Sequence.Sequence_Quality">
<em class="property">class </em><code class="sig-prename descclassname">cardea.fhir.Sequence.</code><code class="sig-name descname">Sequence_Quality</code><span class="sig-paren">(</span><em class="sig-param">dict_values=None</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/cardea/fhir/Sequence.html#Sequence_Quality"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#cardea.fhir.Sequence.Sequence_Quality" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="cardea.fhir.fhirbase.html#cardea.fhir.fhirbase.fhirbase" title="cardea.fhir.fhirbase.fhirbase"><code class="xref py py-class docutils literal notranslate"><span class="pre">cardea.fhir.fhirbase.fhirbase</span></code></a></p>
<p>Raw data describing a biological sequence.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>type</strong> – INDEL / SNP / Undefined variant.</p></li>
<li><p><strong>standardSequence</strong> – Gold standard sequence used for comparing against.</p></li>
<li><p><strong>start</strong> – Start position of the sequence. If the coordinate system is
either 0-based or 1-based, then start position is inclusive.</p></li>
<li><p><strong>end</strong> – End position of the sequence.If the coordinate system is 0-based
then end is is exclusive and does not include the last position. If
the coordinate system is 1-base, then end is inclusive and includes
the last position.</p></li>
<li><p><strong>score</strong> – The score of an experimentally derived feature such as a
p-value
([SO:0001685](http://www.sequenceontology.org/browser/current_svn/term/SO:0001685)).</p></li>
<li><p><strong>method</strong> – Which method is used to get sequence quality.</p></li>
<li><p><strong>truthTP</strong> – True positives, from the perspective of the truth data, i.e.
the number of sites in the Truth Call Set for which there are paths
through the Query Call Set that are consistent with all of the alleles
at this site, and for which there is an accurate genotype call for the
event.</p></li>
<li><p><strong>queryTP</strong> – True positives, from the perspective of the query data, i.e.
the number of sites in the Query Call Set for which there are paths
through the Truth Call Set that are consistent with all of the alleles
at this site, and for which there is an accurate genotype call for the
event.</p></li>
<li><p><strong>truthFN</strong> – False negatives, i.e. the number of sites in the Truth Call
Set for which there is no path through the Query Call Set that is
consistent with all of the alleles at this site, or sites for which
there is an inaccurate genotype call for the event. Sites with correct
variant but incorrect genotype are counted here.</p></li>
<li><p><strong>queryFP</strong> – False positives, i.e. the number of sites in the Query Call
Set for which there is no path through the Truth Call Set that is
consistent with this site. Sites with correct variant but incorrect
genotype are counted here.</p></li>
<li><p><strong>gtFP</strong> – The number of false positives where the non-REF alleles in the
Truth and Query Call Sets match (i.e. cases where the truth is 1/1 and
the query is 0/1 or similar).</p></li>
<li><p><strong>precision</strong> – QUERY.TP / (QUERY.TP + QUERY.FP).</p></li>
<li><p><strong>recall</strong> – TRUTH.TP / (TRUTH.TP + TRUTH.FN).</p></li>
<li><p><strong>fScore</strong> – Harmonic mean of Recall and Precision, computed as: 2 *
precision * recall / (precision + recall).</p></li>
</ul>
</dd>
</dl>
<dl class="method">
<dt id="cardea.fhir.Sequence.Sequence_Quality.assert_type">
<code class="sig-name descname">assert_type</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/cardea/fhir/Sequence.html#Sequence_Quality.assert_type"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#cardea.fhir.Sequence.Sequence_Quality.assert_type" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks class values follow set possible enumerations.</p>
<dl class="field-list simple">
<dt class="field-odd">Raises</dt>
<dd class="field-odd"><p><strong>ValueError</strong> – An error occurs if an attribute of the fhir class
    does not match its possible enumerations.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="cardea.fhir.Sequence.Sequence_Quality.get_relationships">
<code class="sig-name descname">get_relationships</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/cardea/fhir/Sequence.html#Sequence_Quality.get_relationships"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#cardea.fhir.Sequence.Sequence_Quality.get_relationships" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns class relationships.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>A list of the class’s associated relationships.</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="cardea.fhir.Sequence.Sequence_ReferenceSeq">
<em class="property">class </em><code class="sig-prename descclassname">cardea.fhir.Sequence.</code><code class="sig-name descname">Sequence_ReferenceSeq</code><span class="sig-paren">(</span><em class="sig-param">dict_values=None</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/cardea/fhir/Sequence.html#Sequence_ReferenceSeq"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#cardea.fhir.Sequence.Sequence_ReferenceSeq" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="cardea.fhir.fhirbase.html#cardea.fhir.fhirbase.fhirbase" title="cardea.fhir.fhirbase.fhirbase"><code class="xref py py-class docutils literal notranslate"><span class="pre">cardea.fhir.fhirbase.fhirbase</span></code></a></p>
<p>Raw data describing a biological sequence.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>chromosome</strong> – Structural unit composed of a nucleic acid molecule which
controls its own replication through the interaction of specific
proteins at one or more origins of replication
([SO:0000340](http://www.sequenceontology.org/browser/current_svn/term/SO:0000340)).</p></li>
<li><p><strong>genomeBuild</strong> – The Genome Build used for reference, following GRCh build
versions e.g. ‘GRCh 37’.  Version number must be included if a
versioned release of a primary build was used.</p></li>
<li><p><strong>referenceSeqId</strong> – Reference identifier of reference sequence submitted
to NCBI. It must match the type in the Sequence.type field. For
example, the prefix, <em>NG_</em> identifies reference sequence for genes,
<em>NM_</em> for messenger RNA transcripts, and <em>NP_</em> for amino acid
sequences.</p></li>
<li><p><strong>referenceSeqPointer</strong> – A Pointer to another Sequence entity as reference
sequence.</p></li>
<li><p><strong>referenceSeqString</strong> – A string like “ACGT”.</p></li>
<li><p><strong>strand</strong> – Directionality of DNA sequence. Available values are “1” for
the plus strand (5’ to 3’)/Watson/Sense/positive  and “-1” for the
minus strand(3’ to 5’)/Crick/Antisense/negative.</p></li>
<li><p><strong>windowStart</strong> – Start position of the window on the reference sequence.
If the coordinate system is either 0-based or 1-based, then start
position is inclusive.</p></li>
<li><p><strong>windowEnd</strong> – End position of the window on the reference sequence. If
the coordinate system is 0-based then end is is exclusive and does not
include the last position. If the coordinate system is 1-base, then
end is inclusive and includes the last position.</p></li>
</ul>
</dd>
</dl>
<dl class="method">
<dt id="cardea.fhir.Sequence.Sequence_ReferenceSeq.get_relationships">
<code class="sig-name descname">get_relationships</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/cardea/fhir/Sequence.html#Sequence_ReferenceSeq.get_relationships"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#cardea.fhir.Sequence.Sequence_ReferenceSeq.get_relationships" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns class relationships.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>A list of the class’s associated relationships.</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="cardea.fhir.Sequence.Sequence_Repository">
<em class="property">class </em><code class="sig-prename descclassname">cardea.fhir.Sequence.</code><code class="sig-name descname">Sequence_Repository</code><span class="sig-paren">(</span><em class="sig-param">dict_values=None</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/cardea/fhir/Sequence.html#Sequence_Repository"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#cardea.fhir.Sequence.Sequence_Repository" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="cardea.fhir.fhirbase.html#cardea.fhir.fhirbase.fhirbase" title="cardea.fhir.fhirbase.fhirbase"><code class="xref py py-class docutils literal notranslate"><span class="pre">cardea.fhir.fhirbase.fhirbase</span></code></a></p>
<p>Raw data describing a biological sequence.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>type</strong> – Click and see / RESTful API / Need login to see / RESTful API
with authentication / Other ways to see resource.</p></li>
<li><p><strong>url</strong> – URI of an external repository which contains further details
about the genetics data.</p></li>
<li><p><strong>name</strong> – URI of an external repository which contains further details
about the genetics data.</p></li>
<li><p><strong>datasetId</strong> – Id of the variant in this external repository. The server
will understand how to use this id to call for more info about
datasets in external repository.</p></li>
<li><p><strong>variantsetId</strong> – Id of the variantset in this external repository. The
server will understand how to use this id to call for more info about
variantsets in external repository.</p></li>
<li><p><strong>readsetId</strong> – Id of the read in this external repository.</p></li>
</ul>
</dd>
</dl>
<dl class="method">
<dt id="cardea.fhir.Sequence.Sequence_Repository.assert_type">
<code class="sig-name descname">assert_type</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/cardea/fhir/Sequence.html#Sequence_Repository.assert_type"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#cardea.fhir.Sequence.Sequence_Repository.assert_type" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks class values follow set possible enumerations.</p>
<dl class="field-list simple">
<dt class="field-odd">Raises</dt>
<dd class="field-odd"><p><strong>ValueError</strong> – An error occurs if an attribute of the fhir class
    does not match its possible enumerations.</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="cardea.fhir.Sequence.Sequence_Variant">
<em class="property">class </em><code class="sig-prename descclassname">cardea.fhir.Sequence.</code><code class="sig-name descname">Sequence_Variant</code><span class="sig-paren">(</span><em class="sig-param">dict_values=None</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/cardea/fhir/Sequence.html#Sequence_Variant"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#cardea.fhir.Sequence.Sequence_Variant" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="cardea.fhir.fhirbase.html#cardea.fhir.fhirbase.fhirbase" title="cardea.fhir.fhirbase.fhirbase"><code class="xref py py-class docutils literal notranslate"><span class="pre">cardea.fhir.fhirbase.fhirbase</span></code></a></p>
<p>Raw data describing a biological sequence.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>start</strong> – Start position of the variant on the  reference sequence.If the
coordinate system is either 0-based or 1-based, then start position is
inclusive.</p></li>
<li><p><strong>end</strong> – End position of the variant on the reference sequence.If the
coordinate system is 0-based then end is is exclusive and does not
include the last position. If the coordinate system is 1-base, then
end is inclusive and includes the last position.</p></li>
<li><p><strong>observedAllele</strong> – An allele is one of a set of coexisting sequence
variants of a gene
([SO:0001023](http://www.sequenceontology.org/browser/current_svn/term/SO:0001023)).
Nucleotide(s)/amino acids from start position of sequence to stop
position of sequence on the positive (+) strand of the observed
sequence. When the sequence  type is DNA, it should be the sequence on
the positive (+) strand. This will lay in the range between
variant.start and variant.end.</p></li>
<li><p><strong>referenceAllele</strong> – An allele is one of a set of coexisting sequence
variants of a gene
([SO:0001023](http://www.sequenceontology.org/browser/current_svn/term/SO:0001023)).
Nucleotide(s)/amino acids from start position of sequence to stop
position of sequence on the positive (+) strand of the reference
sequence. When the sequence  type is DNA, it should be the sequence on
the positive (+) strand. This will lay in the range between
variant.start and variant.end.</p></li>
<li><p><strong>cigar</strong> – Extended CIGAR string for aligning the sequence with reference
bases. See detailed documentation
[here](<a class="reference external" href="http://support.illumina.com/help/SequencingAnalysisWorkflow/Content/Vault/Informatics/Sequencing_Analysis/CASAVA/swSEQ_mCA_ExtendedCIGARFormat.htm">http://support.illumina.com/help/SequencingAnalysisWorkflow/Content/Vault/Informatics/Sequencing_Analysis/CASAVA/swSEQ_mCA_ExtendedCIGARFormat.htm</a>).</p></li>
<li><p><strong>variantPointer</strong> – A pointer to an Observation containing variant
information.</p></li>
</ul>
</dd>
</dl>
<dl class="method">
<dt id="cardea.fhir.Sequence.Sequence_Variant.get_relationships">
<code class="sig-name descname">get_relationships</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/cardea/fhir/Sequence.html#Sequence_Variant.get_relationships"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#cardea.fhir.Sequence.Sequence_Variant.get_relationships" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns class relationships.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>A list of the class’s associated relationships.</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

</div>


              </div>
              
              
              <div class='prev-next-bottom'>
                

              </div>
              
          </main>
          

      </div>
    </div>

    
  <script src="../_static/js/index.d3f166471bb80abb5163.js"></script>


    <footer class="footer mt-5 mt-md-0">
  <div class="container">
    <p>
          &copy; Copyright 2018, MIT Data To AI Lab.<br/>
        Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.4.4.<br/>
    </p>
  </div>
</footer>
  </body>
</html>